### DISCUSSION TITLE
Can I make a PR to refactor IntegerCastLoop?

### DISCUSSION THREAD

**Discussion by papparapa** [duckdb/duckdb#5459] [General]
I got interested in refactoring `IntegerCastLoop` and I'm working on it.
Currently `IntegerCastLoop` reads all the string in one loop, and handles digits, decimal point, exponent, all of the things in that loop.

https://github.com/duckdb/duckdb/blob/b428e54bd0f77f6fda7993eebfc40340b5b38634/src/common/operator/cast_operators.cpp#L867-L946

I already submitted a PR [#5381](https://github.com/duckdb/duckdb/pull/5381) to fix string with exponent to integer cast (`IntegerCastOperation`), but the code I wrote looks too complex to maintain because I didn't modify IntegerCastLoop and needed to follow its procedure.

After submitting the PR, @Tishj's comments ([this](https://github.com/duckdb/duckdb/issues/5392#issuecomment-1318554713) and [this](https://github.com/duckdb/duckdb/pull/5381#issuecomment-1318247624)) inspired me to think it's a good idea to get the scale information (exponent and decimal point) in string before reading the individual digits.
This would allow us to decide whether to store, round up/down, or ignore the digit when reading it and simplify the code for string  casting.
I need to refactor IntegerCastLoop to do so.

My new PR will split the one big loop in `IntegerCastLoop` into some loops, one handles exponent, one handles decimal point, one handles digits.
I'm going to also refactor each cast operation strust, `IntegerCastOperation`, `HugeIntegerCastOperation` and `DecimalCastOperation` to sort the order of method calls in `IntegerCastLoop`.
It will also make different cast operations to have more unified look.

The PR will add or delete 500+ lines.
Can I continue this work and make the PR?
Please advise.

**Comment by Tishj** [comment]
This would be a question for @Mytherin 
I have held off on this because I suspect this might have a negative effect on performance.

**Comment by Mytherin** [comment]
Thanks for your interest in working on this! Exponents in integers are very rare (and generally not even supported by most systems) - care must be taken to not negatively affect performance of general integer parsing as mentioned by @Tishj. Perhaps it is better to leave this to the core team for now.

**Comment by papparapa** [comment]
Thanks you Tishj and Mytherin!
I understand it could have a negative impact on performance.
I will look for another chance to contribute!
